sdk/storage/azure-storage-common/src/storage_switch_to_secondary_policy.cpp

```diff
@@ -13,7 +13,9 @@ namespace Azure { namespace Storage { namespace _detail {
     SecondaryHostReplicaStatus* replicaStatus = nullptr;
     if (ctx.HasKey(SecondaryHostReplicaStatusKey))
     {
-      replicaStatus = ctx.Get<SecondaryHostReplicaStatus*>(SecondaryHostReplicaStatusKey);
+      replicaStatus
+          = ctx.Get<std::unique_ptr<SecondaryHostReplicaStatus>>(SecondaryHostReplicaStatusKey)
+                .get();
```

Comment:
This issue was introduced in the recent Context changes from https://github.com/Azure/azure-sdk-for-cpp/pull/1824